### ISSUE TITLE
total payment schedule amount validation error

### ISSUE THREAD

**Issue by szufisher** [frappe/erpnext#25950]
## Description of the issue
refer to discussion post https://discuss.erpnext.com/t/total-payment-amount-in-payment-schedule-must-be-equal-to-grand-rounded-total/76428/2

## Context information (for bug reports)

**Output of `bench version`**
```
(paste here)
```

## Steps to reproduce the issue

1.
2.
3.

### Stacktrace / full error message

```
(paste here)
```

## Additional information
erpnext.controllers.accounts_controller
	def validate_payment_schedule_amount(self):

![image](https://user-images.githubusercontent.com/12823863/120740298-89b90d00-c525-11eb-9b47-d862a61b1d9a.png)

```
			if flt(total - grand_total, self.precision("grand_total")) != 0 or \
				flt(base_total - base_grand_total, self.precision("base_grand_total")) != 0:
				frappe.throw(_("Total Payment Amount in Payment Schedule must be equal to Grand / Rounded Total"))
```

OS version / distribution, `ERPNext` install method, etc.

**Comment by git529** [comment]
Hi. Has this bug been fixed?

**Comment by Ponyooooo** [comment]
Hi, do you have any updates on this? we are currently experiencing this on the Quotation doctype after we migrated from v12 to v13.

**Comment by Ponyooooo** [comment]
This error/bug was solved on the v12, https://github.com/frappe/erpnext/issues/12057:  Why is the v13 has the same bug again?

**Comment by centaur78** [comment]
Hi all

This bug has again resurfaced with the latest bench update 

This validation seems to have changed in April 2021 ..... this PR #25292 specifically this https://github.com/frappe/erpnext/pull/25292/commits/d552fe6778f71e940f4acdb594afcf79fe5757c2

But why the error now again??

**Comment by cacarios** [comment]
Hi all,

Is there a fix for it? Would an "if invoice portion is 0 and payment amount is > 0, then get payment amount, divide by rounded total, multiply by 100 then value to invoice portion" function triggered on before_save or validation work?

Does anyone have another solution?

**Comment by cacarios** [comment]
Hi @yadavyk could you please validate this issue?

**Comment by HarryPaulo** [comment]
@Ponyooooo this bug still happend...

My situation.

**sales order / sales invoice**

i'm selling 1 item, sale price is 31,31.

If i  use payments terms with 2x of 50% each my payments schedule will show 15,16 each term.

![image](https://user-images.githubusercontent.com/25017988/138710396-9d88114a-9930-46bd-b67e-c302b7c36bb4.png)

15,16 + 15,16 = 31,32, then i get the same message: **Total Payment Amount in Payment Schedule must be equal to Grand / Rounded Total**


this code must be improved to the last record on on payment schedule get the rest of the invoice amount.

Example: 
1 - 15,16
2 - 15,15

**Comment by git529** [comment]
We are still experiencing this bug. Very annoying. Hopefully someone can fix it.

**Comment by HarryPaulo** [comment]
@git529 i got a solution for this PR, i will create a pull request tomorrow

**Comment by git529** [comment]
> @git529 i got a solution for this PR, i will create a pull request tomorrow

Looking forward to it. Thanks.

**Comment by git529** [comment]
We are also experiencing same issue on Purchase Order

**Comment by HarryPaulo** [comment]
i did a revert, but u can see the source code and copy it, i got another pull request open, i need it must be approved before create a pull request for this problem

**Comment by lafi87** [comment]
Any update on this?

**Comment by git529** [comment]
> Any update on this?

The issue was fixed for us when we updated to the next version.
